src/main.rs

```diff
@@ -62,29 +142,40 @@ fn main() -> Result<(), Error> {
         }
     };
 
-    let mut map: Map = Map::new(header);
+    let mut map: Map<V> = Map::new(header);
+
+    fn process_change<T, V: FromValue<T> + ValueType>(
+        map: &mut Map<V>,
+        id: IdCode,
+        value: T,
+    ) -> Result<(), Error> {
+        if let Some(value) = V::from_value(value)? {
+            map.change_value(id, value)?;
+        }
+        Ok(())
+    }
 
     for cmd in commands {
         match cmd? {
             Command::Timestamp(_t) => {
                 map.push_timestep();
             }
-            Command::ChangeScalar(id, x) => map.change_value(
-                id,
-                u64::from_str_radix(&to_bitstring(&[x]), 2)?,
-            )?,
-            Command::ChangeVector(id, x) => map
-                .change_value(id, u64::from_str_radix(&to_bitstring(&x), 2)?)?,
+            Command::ChangeScalar(id, x) => process_change(&mut map, id, x)?,
+            Command::ChangeVector(id, x) => process_change(&mut map, id, x)?,
+            Command::ChangeReal(id, x) => process_change(&mut map, id, x)?,
+            Command::ChangeString(id, x) => process_change(&mut map, id, x)?,
             _ => (),
         }
     }
 
     let mut output = Output::default();
 
-    for (path, var) in map.paths {
-        output.insert(path, &map.items[&var.code]);
+    let Map { mut items, paths } = map;
+
+    for (path, var) in paths {
+        output.insert(path, items.remove(&var.code).unwrap().values);
```

Comment:
The remove here means that if `&var.code` shows up multiple times, the second instance will crash vcdump. Change this to `get()`

Reply:
fixed